api.py

```diff
@@ -75,17 +75,16 @@ def initialize_app():
             ),
             (
                 r"/api/v1/grading_run/{}/{}".format(
-                    ID_REGEX.format("course_id"), ID_REGEX.format("assignment_name")
+                    ID_REGEX.format("course_id"),
+                    ID_REGEX.format("assignment_name_or_run_id"),
```

Comment:
Can you explain this change? Making the URL parameter take on multiple interpretations (assignment name sometimes, run_id other times) doesn't quite feel right to me.

Reply:
Ayush decided to remove <assignment name> from GET /grading_run. But then the pattern of POST /grading_run overlaps with GET and I didn't find a good way to write patterns for routing so...

Reply:
@ayushr2 please elaborate

Reply:
Oh, this route is not supposed to change. I think the issue defines a new endpoint `/api/v1/grading_run_status` for getting statuses and `/api/v1/grading_job_log` for job logs.

This endpoint to should remain unchanged

Reply:
`/api/v1/grading_run_status` does not need the assignment parameter. It just authenticates the course and gets the grading run status if and only if that run id belongs to a grading run started by that course.

`/api/v1/grading_run` **needs** the assignment because it is being used to start AG runs. 

We need to make a new `APIHandler` class in `handlers/client` for getting the status

Reply:
ah ok fixed